### DISCUSSION TITLE
How to remove the password from the p12 private key file with openssl?

### DISCUSSION THREAD

**Discussion by kustodian** [fastlane/fastlane#22114] [Q&A]
Because we don't have access to the App Store portal we were sent certificates, keys and provisioning profile files. Because fastlane match import requires to use p12 private keys without a password we removed the password by importing the key into the Keychain and exporting it without a key, but anything we tried with OpenSSL it didn't want to work. We tried everything from this https://serverfault.com/questions/515833/how-to-remove-private-key-password-from-pkcs12-container discussion, but nothing helped. Does anybody know what `openssl` commands can do the same thing that the Keychain import/export does?

**Comment by kustodian** [comment]
Nobody? Is it possible that nobody but us have this issue?

**Comment by wow-such-amazing** [comment]
@kustodian I'm struggling with this as well. Eventually was able to import certificates to our repo. And match didn't even show any warnings when running the build. But then CodeSign command fails for no particular reason 😕 

Gonna try to import a pem file as mentioned in this discussion https://github.com/fastlane/fastlane/discussions/19814 🤞

**Comment by lichtmetzger** [comment]
I got a .p12 file from a customer which had the added bonus of using an outdated security standard and I needed to convert this into a .crt and .key file for Apache/nginx.

First, I tried removing the passphrase like this:
```
openssl pkcs12 -in original.p12 -nodes -out temp.pem
openssl pkcs12 -export -in temp.pem -out unprotected.p12
```

That errored out, though:
```
8032648CFE7E0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:355:Global default library context, Algorithm (RC2-40-CBC : 0), Properties ()
```

The customer used `RC2-40-CBC`, which OpenSSL doesn't like, but with adding `-legacy`, it does:
```
openssl pkcs12 -in original.p12 -legacy -nodes -out temp.pem
openssl pkcs12 -export -in temp.pem -legacy -out unprotected.p12
```

Now we have an unprotected.p12 without a passphrase. To convert it for webservers, just use this:
```
openssl pkcs12 -in unprotected.p12 -legacy -nocerts -nodes -out web.key
openssl pkcs12 -in unprotected.p12 -legacy -clcerts -nokeys -nodes -out web.crt
```

**Comment by jyotirmaygupta23-tech** [comment]
Integrating ELM Absher OTP via SOAP sendOTPWithDynamicTemplate to https://otp.absher.sa/AbsherOTPService (AOTPSIG v1.6). SOAP creds and template are configured (clientId, clientAuthorization, MedicalCD-OTP-01, etc.).

Blocker: ELM sent medicalcloud.p12 (from cert renewal email) but we don’t have the P12 export password and can’t obtain it from the client right now. OpenSSL: Mac verify error: invalid password? with empty pass. Without unlocking the P12, Windows curl fails mTLS (HTTP 000 / Schannel handshake error) before any SOAP response.

Have: medicalcloud.p12, medicalcloud.crt/.cer (public cert only — no private key).
Need: Way to use mTLS for this client cert without the original P12 password, or an alternative ELM-supported approach.

Not using: Sadq KYB API (sandbox-api.sadq-sa.com) — our integration is direct ELM SOAP + mTLS.

Any recommended path (e.g. cert reissue, PEM key from ELM, server-side export if cert is already installed elsewhere)?

@lichtmetzger @kustodian if you guys can provide some solution.
